### ISSUE TITLE
Why the training is too slow.

### ISSUE THREAD

**Issue by RaceSu** [amdegroot/ssd.pytorch#148]

**Comment by RaceSu** [comment]
I use 4 gpu

**Comment by majuncai** [comment]
i have the same problem ,how do you solve it?

**Comment by Godricly** [comment]
All of the loss computation is done on GPU 0. You can notice much more memory cunsumption on that card. With a batch size of 128, you will call match function 128 times on GPU 0. In my customed case, it took 1 sec in the loss.
